### PR TITLE
Added references-plot_dist

### PR DESCRIPTION
**Pull request by roshnaeem**
Description:
Added references to the docstring of [plot_dist](https://arviz-devs.github.io/arviz/api/generated/arviz.plot_dist.html).

Fixes: #1824

### CODE REVIEW

File: arviz/plots/distplot.py [hunk 1]
```diff
@@ -80,24 +80,37 @@ def plot_dist(
...
-        Keywords passed to the histogram.
+        Keyword arguments used to customize the histogram. Ignored when plotting a KDE.
+        They are passed to :meth:`matplotlib.axes.Axes.hist` if using matplotlib,
+        or to :meth:`bokeh.plotting.Figure.quad` if using bokeh. In bokeh case,
+        the following extra keywords are also supported:
+
+        * ``color``: replaces the ``fill_color`` and ``line_color`` of the ``quad`` method
+        * ``bins``: taken from ``hist_kwargs`` and passed to :func:`numpy.histogram` instead
+        * ``density``: normalize histogram to represent a probability density function,
+        Defaults to ``True``
```
Comment (OriolAbril) [inline arviz/plots/distplot.py:110]: The indentation is not right and renders the default as a new paragraph outside the list. It should be at the same level as the start of the text in the bullet point, not at the same level as the bullet point icon

Comment (roshnaeem) [reply arviz/plots/distplot.py:110]: I tried adding indentation but it was failing the checks and messing up the UI.

Comment (OriolAbril) [reply arviz/plots/distplot.py:110]: how many spaces?

Comment (roshnaeem) [reply arviz/plots/distplot.py]: Done, please check.

**Pull request merged**